### PR TITLE
Create HexNode.java

### PR DESCRIPTION
**Pull request by jaycoug**

### CODE REVIEW

**Review by dwilliams56 — APPROVED** [review]

File: src/main/Boards/HexNode.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public class HexNode {
+	
+	private int nodeID;
+	private int value;
+	//For red player, node value = 1; Blue player = 2.
+	
+	public HexNode(int nodeID) {
+		this.nodeID = nodeID;
+		this.value = 0;
+	}
```
Comment (djimenezmsu) [inline src/main/Boards/HexNode.java:17]: If value represents the color of the player (1, 2) why should the constructor without a value passed set the value to 0 (not a valid color)? As a group we should decide if the default color of the player should be red or blue in my opinion. Setting it to default as 0 doesn't make sense to me.

**Review by djimenezmsu — CHANGES_REQUESTED** [review]
See above comment

**Review by mrichardson3065 — APPROVED** [review]

File: src/main/Boards/HexNode.java (hunk 1 above)
Comment (jaycoug) [reply src/main/Boards/HexNode.java:17]: This constructor follows the Trevorah project's constructor in the HexLocation class for setting the node ID without inputting a value for the player. (Setting a specific player is handled in the constructor below this one.) The 0 comes from the Board.BLANK value in the Board interface (which I didn't use, but we can if we want.) I think setting the value to 0 just indicates that the tile hasn't been selected by a player, and sets the color/status to white/unselected. This looks to be important for when the board is initialized with unselected tiles in the initBoard() method of the BoardData class. We can try it without, though, if it still seems extraneous.

Comment (djimenezmsu) [reply src/main/Boards/HexNode.java:17]: Ok, thanks for the explanation

**Review by djimenezmsu — APPROVED** [review]

**Pull request merged**